### ISSUE TITLE
`CallImplicitGemmWrwDynamic` needs refactoring

### ISSUE THREAD

**Issue by atamazov** [ROCm/MIOpen#756]
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/eec7091fbad407e9d43eae8985ad8eb4e0861689/src/solver/conv_asm_implicit_gemm_wrw_gtc_dynamic_xdlops.cpp#L230

All possible computations to be moved to Factory. The Solver lambda shall only hold copies of values pre-computed in Factory. No memory allocations/deallocations, no function calls etc.

**Comment by shaojiewang** [comment]
OK. I will do that with PR#691

**Comment by atamazov** [comment]
@shaojiewang 
> OK. I will do that with #691

Is it possible to do that in separate PR?

**Comment by shaojiewang** [comment]
> @shaojiewang
> 
> > OK. I will do that with #691
> 
> Is it possible to do that in separate PR?

Yes, possible. It will be done with a separate PR.

**Comment by atamazov** [comment]
Depends on #691

> And `CallImplicitGemmWrwDynamic` will be refactored by a separate PR.

_Originally posted by @shaojiewang in https://github.com/ROCmSoftwarePlatform/MIOpen/issues/691#issuecomment-816368176_

All the parameters that depend only on problem config should be computed in the Factory and passed to the Invoker (well, captured by it) by value.

If only one kernel us used, no need to use a vector of kernel objects. Manual adjustments of kernel time are not necessary in this case as well.

Example or refactoring: https://github.com/ROCmSoftwarePlatform/MIOpen/pull/680/commits/6d9cf43ce3e7ee24fb6c63fe8c97ecd6fbd586d4

**Comment by atamazov** [comment]
Moved to 4.4. This is our internal goal.

**Comment by atamazov** [comment]
Moved to 5.1

**Comment by ppanchad-amd** [comment]
@atamazov Is this resolved on ROCm 6.0.2? If so, please close ticket. Thanks!

**Comment by atamazov** [comment]
@ppanchad-amd No, this is not resolved yet. Current code: https://github.com/ROCm/MIOpen/blob/cee33a7aaa3dbbe850d2580a72fff8774ddbc131/src/solver/conv_asm_implicit_gemm_wrw_v4r1_dynamic.cpp#L74

**Comment by JonathanLichtnerAMD** [comment]
Closing since MIOpen is being migrated to [ROCm/rocm-libraries](https://github.com/ROCm/rocm-libraries).  Please re-open there if you believe this is still an issue.
